src/sentry/api/helpers/group_index/update.py

```diff
@@ -537,149 +531,29 @@ def update_groups(
-                            sender=update_groups,
-                        )
-            elif new_status == GroupStatus.IGNORED:
-                activity_type = ActivityType.SET_IGNORED.value
-                activity_data = {
-                    "ignoreCount": ignore_count,
-                    "ignoreDuration": ignore_duration,
-                    "ignoreUntil": ignore_until,
-                    "ignoreUserCount": ignore_user_count,
-                    "ignoreUserWindow": ignore_user_window,
-                    "ignoreWindow": ignore_window,
-                }
-
-                groups_by_project_id = defaultdict(list)
-                for group in group_list:
-                    groups_by_project_id[group.project_id].append(group)
-
-                for project in projects:
-                    project_groups = groups_by_project_id.get(project.id)
-                    if project_groups:
-                        issue_ignored.send_robust(
-                            project=project,
-                            user=acting_user,
-                            group_list=project_groups,
-                            activity_data=activity_data,
-                            sender=update_groups,
-                        )
-
-            for group in group_list:
-                group.status = new_status
-
-                activity = Activity.objects.create(
-                    project=project_lookup[group.project_id],
-                    group=group,
-                    type=activity_type,
-                    user_id=acting_user.id,
-                    data=activity_data,
-                )
-                record_group_history_from_activity_type(group, activity_type, actor=acting_user)
-
-                # TODO(dcramer): we need a solution for activity rollups
-                # before sending notifications on bulk changes
-                if not is_bulk:
-                    if acting_user:
-                        GroupSubscription.objects.subscribe(
-                            user=acting_user,
-                            group=group,
-                            reason=GroupSubscriptionReason.status_change,
-                        )
-                    activity.send_notification()
-
-                if new_status == GroupStatus.UNRESOLVED:
-                    kick_off_status_syncs.apply_async(
-                        kwargs={"project_id": group.project_id, "group_id": group.id}
-                    )
+        if group_list and status_updated:
+            activity_type, activity_data = handle_status_update(
+                group_list=group_list,
+                projects=projects,
+                project_lookup=project_lookup,
+                new_status=new_status,
+                is_bulk=is_bulk,
+                acting_user=acting_user,
+                status_details=result["statusDetails"],
+                sender=update_groups,
+                activity_type=activity_type,
```

Comment:
future refactor may be able to simplify the parameter list. This PR only moves code, no logic should be altered